### DISCUSSION TITLE
cFS projects that has used CSP

### DISCUSSION THREAD

**Discussion by GateraGael** [nasa/cFS#397] [Q&A]
Hello, I would like to ask if there are any cFS projects that have used the CubeSat Protocol.
https://github.com/libcsp/libcsp

If any please have any references, and if an open-source with links to documentation.

**Comment by ghost** [comment]
No implementations that I am aware of - I used it on a Cubesat to copy cFS files between a development host and the on board computer (Gomspace Nanomind) but it was not part of the flight software.

**Comment by sf314** [comment]
One project, the [Phoenix CubeSat](https://phxcubesat.asu.edu) intended to use CSP for communications between the ground and the CubeSat, and as well as between the onboard computer and radio module. CSP packets would route from a client running in CFS, to the radio module, then transmitted to a CSP server on the ground station.

If you're asking about how to integrate CSP into a CFS app, I believe the root CFS Makefile should be updated in order to build `libcsp.a` before the rest of CFS. Then you can reference `libcsp.a` when building the apps that use it.

**Reply by jeffstjean** [reply]
Did you manage to get this working? In older versions of libcsp, you can add `-fPIC` to the build flags to get position independent code or use the `--enable-shlib` in newer versions. How did you link it properly?
